src/qirqsim/qsimQuantum.cc

```diff
@@ -0,0 +1,288 @@
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//---------------------------------------------------------------------------//
+//! \file qirxacc/qsimQuantum.cc
+//---------------------------------------------------------------------------//
+
+#include "qsimQuantum.hh"
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+#include <thread>
+#include <utility>
+
+#include "qiree/Assert.hh"
+
+// Qsim
+#include <qsim/lib/circuit.h>
+#include <qsim/lib/circuit_qsim_parser.h>
+#include <qsim/lib/formux.h>
+#include <qsim/lib/fuser.h>
+#include <qsim/lib/fuser_mqubit.h>
+#include <qsim/lib/gate.h>
+#include <qsim/lib/gates_qsim.h>
+#include <qsim/lib/io.h>
+#include <qsim/lib/io_file.h>
+#include <qsim/lib/run_qsim.h>
+#include <qsim/lib/simmux.h>
+#include <qsim/lib/simulator_basic.h>
+#include <qsim/lib/statespace_basic.h>
+#include <qsim/lib/util_cpu.h>
+//
+
+namespace qiree
+{
+//---------------------------------------------------------------------------//
+/*
+Initialize the qsim simulator
+*/
+
+qsimQuantum::State qsimQuantum::init_state_space()
+{  // check if StateSpace is the proper type for the output, problably it is
+   // just State from the Fatory struct.
+    std::srand(static_cast<unsigned int>(std::time(nullptr)));  // Seed the
+                                                                // random
+                                                                // number
+                                                                // generator
+    qsimParam.seed = std::rand();  // Set the seed for qsim parameters
+    numThreads = std::max(
+        1, static_cast<int>(std::thread::hardware_concurrency()));  // Get the
+                                                                    // number
+                                                                    // of
+                                                                    // threads
+    qsimParam.max_fused_size = 2;  // Set the maximum size of fused gates
+    qsimParam.verbosity = 0;  // see verbosity in run_qsim.h
+    // Initialize the qsim simulator
+    qsimQuantum::StateSpace state_space
+        = Factory(numThreads).CreateStateSpace();  // Create the state space
+    State state = state_space.Create(this->num_qubits());  // Create the state
+    // Check if the state is null
+    if (state_space.IsNull(state))
+    {
+        qsim::IO::errorf(
+            "not enough memory: is the number of qubits too large?\n");
+    }
+    state_space.SetStateZero(state);  // Set the state to zero, TODO: the
+                                      // initial state is not necessarily zero
+    return state;
+}
+
+qsimQuantum::qsimQuantum(std::ostream& os, size_type shots) : output_(os) {}
+
+//---------------------------------------------------------------------------//
+/*
+Prepare to build a quantum circuit for an entry point
+*/
+
```

Comment:
Formatting needed here and elsewhere
```suggestion
/*!
 * Prepare to build a quantum circuit for an entry point.
 */

```

Reply:
Resolved in [commit 22d0ac9](https://github.com/ORNL-QCI/qiree/pull/17/commits/22d0ac9880b07fb331beb296b9b7a2b923d45f65).